File: src/game/systems/container.lua
local Aspect = require 'ecs.aspect'
local System = require 'ecs.system'
local Container = require 'game.components.container'

local aspect = Aspect:new({Container})
local ContainerSystem = System:new('container', aspect)

function ContainerSystem:update(dt)
  for _, entity in pairs(self.entities) do
  end
end

return ContainerSystem

File: packages/client/src/game/systems/Container.ts
import { Aspect } from '../../ecs/Aspect'
import { System } from '../../ecs/System'
import { Container } from '../components/Container'
import { SystemFlag } from '../flags'

export class ContainerSystem extends System {
  static _id = 'Container'
  _id = ContainerSystem._id

  static _flag = SystemFlag.Container
  _flag = ContainerSystem._flag

  static _aspect = new Aspect([Container])
  _aspect = ContainerSystem._aspect

  update = (dt: number) => {
    this.entities.forEach(entity => {})
  }
}
